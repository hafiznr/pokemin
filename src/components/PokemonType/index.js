import { Box } from "@chakra-ui/react";

const CUSTOM_BG = {
  normal: '#aa9',
  fire: '#f42',
  water: '#39f',
  electric: '#fc3',
  grass: '#7c5',
  ice: '#6cf',
  fighting: '#b54',
  poison: '#a59',
  ground: '#db5',
  flying: '#89f',
  psychic: '#f59',
  bug: '#ab2',
  rock: '#ba6',
  ghost: '#66b',
  dragon: '#76e',
  dark: '#754',
  steel: '#aab',
  fairy: '#e9e',
};

const PokemonType = ({type}) => {
  
  return (
    <Box
      padding="6px 2px"
      width="76px"
      textTransform="uppercase"
      textAlign="center"
      bgColor={CUSTOM_BG[type]}
      borderRadius="4px"
      color="white"
      textShadow="1px 1px 2px rgb(0 0 0 / 70%)"
      mb="4px"
      data-testid="pokemon-type"
    >
      {type}
    </Box>
  )
}

export default PokemonType;