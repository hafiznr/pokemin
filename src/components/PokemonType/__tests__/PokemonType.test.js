import { render, screen } from '@testing-library/react';
import PokemonType from '../';

test('renders PokemonType', () => {
  const { container, rerender } = render(<PokemonType/>);

  expect(container).toMatchSnapshot();

  rerender(<PokemonType type="psychic" />)
  expect(screen.getByTestId('pokemon-type')).toHaveTextContent('psychic');
});