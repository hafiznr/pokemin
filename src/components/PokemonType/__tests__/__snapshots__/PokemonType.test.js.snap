// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokemonType 1`] = `
<div>
  <div
    class="css-1r25m91"
    data-testid="pokemon-type"
  />
</div>
`;
